commit 388d6bcccfa46f6b9fa170c97ec2e3c279b3f961
Author: Eric Anholt <eric@anholt.net>
Date:   Mon May 9 17:28:18 2016 -0700

    clk: bcm2835: Mark GPIO clocks enabled at boot as critical.
    
    These divide off of PLLD_PER and are used for the ethernet and wifi
    PHYs source PLLs.  Neither of them is currently represented by a phy
    device that would grab the clock for us.
    
    This keeps other drivers from killing the networking PHYs when they
    disable their own clocks and trigger PLLD_PER's refcount going to 0.
    
    v2: Skip marking as critical if they aren't on at boot.
    
    Signed-off-by: Eric Anholt <eric@anholt.net>

diff --git a/drivers/clk/bcm/clk-bcm2835.c b/drivers/clk/bcm/clk-bcm2835.c
index e1ceafdc7dce..b6399a4fc53f 100644
--- a/drivers/clk/bcm/clk-bcm2835.c
+++ b/drivers/clk/bcm/clk-bcm2835.c
@@ -1443,6 +1443,15 @@ static struct clk_hw *bcm2835_register_clock(struct bcm2835_cprman *cprman,
 	init.name = data->name;
 	init.flags = data->flags | CLK_IGNORE_UNUSED;
 
+	/*
+	 * Some GPIO clocks for ethernet/wifi PLLs are marked as
+	 * critical (since some platforms use them), but if the
+	 * firmware didn't have them turned on then they clearly
+	 * aren't actually critical.
+	 */
+	if ((cprman_read(cprman, data->ctl_reg) & CM_ENABLE) == 0)
+		init.flags &= ~CLK_IS_CRITICAL;
+
 	/*
 	 * Pass the CLK_SET_RATE_PARENT flag if we are allowed to propagate
 	 * rate changes on at least of the parents.
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1022_388d6bcccfa46f6b9fa170c97ec2e3c279b3f961.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1022_388d6bcccfa46f6b9fa170c97ec2e3c279b3f961.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
